src/lib/storage/disks_list.rb

```diff
@@ -0,0 +1,58 @@
+#
+# encoding: utf-8
+
+# Copyright (c) [2016] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "storage"
+require "storage/devices_list"
+require "storage/partitions_list"
+require "storage/filesystems_list"
+#require "storage/free_disk_spaces_list"
+require "storage/refinements/disk"
+
+module Yast
+  module Storage
+    class DisksList < DevicesList
+      list_of ::Storage::Disk
+      by_default_delegate_to :partitions
+
+      using Refinements::Disk
+
+      def partitions
+        part_list = list.reduce([]) { |sum, disk| sum + disk.all_partitions }
```

Comment:
this can be performance problem if there are thousands of disks and each have some partitions ( yes, we have such case in past and it took 10 hours to install it ). so better use `sum.concat(disk.all_partitions)`